src/core/meshcop/dataset.cpp

```diff
@@ -487,7 +482,9 @@ Error Dataset::AppendMleDatasetTlv(Type aType, Message &aMessage) const
         else if (cur->GetType() == Tlv::kDelayTimer)
         {
             uint32_t      elapsed = TimerMilli::GetNow() - mUpdateTime;
-            DelayTimerTlv delayTimer(static_cast<const DelayTimerTlv &>(*cur));
+            DelayTimerTlv delayTimer;
+
+            delayTimer = *As<DelayTimerTlv>(cur);
```

Comment:
nit: why not reduce to a single line?
```suggestion
            DelayTimerTlv delayTimer = *As<DelayTimerTlv>(cur);
```